### PR TITLE
feat(auth-kit): SafeAuthPack

### LINKED ISSUE #565: [Web3Auth Safe Signer] Create SafeAuthPack
## Context / issue
After the initial evaluation #539 of the new Web3Auth SDK and obtaining positive results, the next step is to implement the final integration in the `safe-core-sdk` repository.

## Proposed solution
- [x] Removing the current `Web3AuthModalPack`
- [x] Create a new `SafeAuthPack` using the new SDK (`@web3auth/ws-embed`) following the `AuthKitBasePack` interface
- [x] Add a new required `isAuthenticated` property to the `AuthKitBasePack` pack in order to make it easier to devs to identify this status
- [x] Update `auth-kit` example app
- [x] Update `onramp-kit` example app

## Additional context
[Web3Auth Safe Signer epic](https://www.notion.so/safe-global/Web3auth-Safe-Signer-b3d6c72d3a3047408d992a0130c7d642)

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves #565 

### auth-kit
- Remove `Web3AuthNoModalPack` and add `SafeAuthPack` with the same api

### CODE REVIEW

File: packages/auth-kit/example/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "@emotion/styled": "^11.10.5",
     "@mui/material": "^5.11.7",
     "@safe-global/safe-react-components": "^2.0.1",
-    "ethers": "^6.0.3",
+    "@toruslabs/ethereum-controllers": "^4.5.2",
+    "ethers": "^5.7.2",
```
Comment (dasanra) [inline packages/auth-kit/example/package.json:17]: Why this rollback?

Comment (mrmuhammad835) [reply packages/auth-kit/example/package.json:18]: I need help stuck in finding my digital identity

File: packages/protocol-kit/src/contracts/utils.ts [hunk 2]
```diff
@@ -161,7 +161,7 @@ const memoizedGetProxyCreationCode = createMemoizedFunction(
  * @param {number} chainId - The chain ID associated with the chain.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/contracts/utils.ts:161]:
```suggestion
 * @param {bigint} chainId - The chain ID associated with the chain.
```

File: packages/onramp-kit/example/client/package.json [hunk 3]
```diff
@@ -5,20 +5,23 @@
...
-    "@web3auth/openlogin-adapter": "^4.6.0",
     "ethers": "^6.7.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "react-router-dom": "^6.18.0"
+    "react-router-dom": "^6.18.0",
+    "vite-plugin-node-polyfills": "^0.16.0"
   },
   "devDependencies": {
     "@esbuild-plugins/node-globals-polyfill": "^0.2.3",
```
Comment (dasanra) [inline packages/onramp-kit/example/client/package.json:27]: In the other example app you removed this devDependencies

Comment (yagopv) [reply packages/onramp-kit/example/client/package.json]: fixed

**Review by dasanra — APPROVED** [review]

**Pull request merged**